### PR TITLE
Add `Keys` method to EnvMap

### PR DESCRIPTION
**Pull request by tcahill**

### CODE REVIEW

**Comment by tcahill** [comment]
@krobertson

File: envmap/envmap.go [hunk 1]
```diff
@@ -114,12 +114,21 @@ func (e *EnvMap) Map() map[string]string {
 }
 
 func (e *EnvMap) Strings() []string {
-	m := e.Map()
-	r := make([]string, 0, len(m))
```
Comment (krobertson) [inline envmap/envmap.go]: Why change this? The benefit of how it was, was that it knows how big the array will be and pre-allocates it.

**Comment by krobertson** [comment]
General LGTM. Pre-allocating the memory is a bit nicer, so would be good to keep that in the two functions.

Failure was in s3util with a "network unreachable" error...

**Comment by tcahill** [comment]
Reverted the change to `Map()`.

**Comment by tcahill** [comment]
Fixed the S3 upload test that was failing: the server was listening on IPv6 loopback, but it seems the string representation was not correct (was getting rendered like `[::]:33102`). Listening on `127.0.0.1:0` resolves this.

**Pull request merged**